homeassistant/components/lastfm/sensor.py

```diff
@@ -105,25 +105,27 @@ def __init__(self, user: User, entry_id: str) -> None:
     def update(self) -> None:
         """Update device state."""
         try:
-            self._user.get_playcount()
-        except WSError as exc:
+            play_count = self._user.get_playcount()
+        except PyLastError as exc:
             self._attr_available = False
             LOGGER.error("Failed to load LastFM user `%s`: %r", self._user.name, exc)
             return
-        self._attr_entity_picture = self._user.get_image()
-        if now_playing := self._user.get_now_playing():
-            self._attr_native_value = format_track(now_playing)
-        else:
-            self._attr_native_value = STATE_NOT_SCROBBLING
-        top_played = None
-        if top_tracks := self._user.get_top_tracks(limit=1):
-            top_played = format_track(top_tracks[0].item)
-        last_played = None
-        if last_tracks := self._user.get_recent_tracks(limit=1):
-            last_played = format_track(last_tracks[0].track)
-        play_count = self._user.get_playcount()
-        self._attr_extra_state_attributes = {
-            ATTR_LAST_PLAYED: last_played,
-            ATTR_PLAY_COUNT: play_count,
-            ATTR_TOP_PLAYED: top_played,
-        }
+        try:
+            self._attr_entity_picture = self._user.get_image()
+            if now_playing := self._user.get_now_playing():
+                self._attr_native_value = format_track(now_playing)
+            else:
+                self._attr_native_value = STATE_NOT_SCROBBLING
+            top_played = None
+            if top_tracks := self._user.get_top_tracks(limit=1):
+                top_played = format_track(top_tracks[0].item)
+            last_played = None
+            if last_tracks := self._user.get_recent_tracks(limit=1):
+                last_played = format_track(last_tracks[0].track)
+            self._attr_extra_state_attributes = {
+                ATTR_LAST_PLAYED: last_played,
+                ATTR_PLAY_COUNT: play_count,
+                ATTR_TOP_PLAYED: top_played,
+            }
+        except PyLastError:
+            return
```

Comment:
This feels like a large try scope? As in, if `get_top_tracks` fails, a user is left with a partial update? While `get_recent_tracks` isn't even tried anymore?

Reply:
When I look at my logs this happened like 20 times in the last 4 days. What should the preferred option be? In my opinion, the best way would be to split the data up into several sensors so we can put it unavailable independently, but that'd be a breaking change. So would putting everything in one try except be the best solution for now?

Reply:
I think I would implement the `available` state right now, and make the sensor unavailable. As a follow-up PRs:

- Implement a data coordinator
- Add sensor descriptions
- Add separate sensors for the attributes
- Keep the attributes for the entity around for a couple of release cycles before removing them from the old sensor entity.